feat: add `search` section to `landing page`

enzo-bitfly opened the pull request:
See: FE-92

enzo-bitfly commented on frontend/layers/products/app/components/BlockchainSearchResultItem.vue:
```diff
@@ -0,0 +1,125 @@
+  <NuxtLink
+    v-if="result.type === 'address'"
+    class="flex gap-md"
+    :to="`${baseUrl}/address/${result.value.address.hash}?chain_id=${result.chain_id}`"
```
I decided to add the `baseUrl` here and skip using the `global redirect middleware` altogether, since otherwise I'd have needed to pass the `chain_id` as a query param for the middleware to handle, and make everything worse there. 
Open to discuss, preferably after merging this.

marcel-bitfly commented on frontend/layers/base/app/assets/css/main.css:
```diff
@@ -47,8 +52,9 @@
   --container-3xs: 16rem; /* 256px */
   /* --container-7xl: 80rem; 1280px -> this is in conflict with --spacing-7xl */
   --container-8xl: 90rem; /* 1440px */
+  --container-24xl: 80rem; /* 1280px */
```
**note**: we should be able to remove `--spacing-7xl` and so on with `--spacing: 0.25rem;` now

which will mean we do not need these high `xl values` anymore
(this could be done in a separate pr though)

marcel-bitfly commented on frontend/layers/base/app/components/BaseNetworkIcon.vue:
```diff
@@ -0,0 +1,23 @@
+import { useNetwork } from '~/layers/base/app/composables/useNetwork'
+
+const { id } = defineProps<{
+  id: ChainId,
```
** thought**(non-blocking): maybe being more explicit here would be better? 

I probably do not know what you mean with `id of BaseNetworkIcon` (could also be an internal thing)


```suggestion
  chainId: ChainId,
```

enzo-bitfly replied:
I agree. Changed to `chainId`

marcel-bitfly commented on frontend/layers/base/app/components/BaseNetworkIcon.vue:
```diff
@@ -0,0 +1,23 @@
+    class="w-2xl"
+  />
+  <LazyNetworkIconCryptoHoodi
+    v-if="networkName === 'Ethereum Hoodi'"
```
**np**: maybe `short name` is more convenient here? 
```suggestion
    v-if="shortName === 'Hoodi'"
```

enzo-bitfly replied:
Done, added a utility for this in `layers/base/utils/networks.ts`

marcel-bitfly commented on frontend/layers/base/app/components/BaseSearchInput.vue:
```diff
@@ -0,0 +1,168 @@
+    }
+  },
+  {
+    deep: true,
```
**question**: are you sure this is needed, it should be the `default`?

See: https://vuejs.org/guide/essentials/watchers#deep-watchers

enzo-bitfly replied:
Good catch. This is a residue from having the model as an earlier implementation. Removed.

marcel-bitfly commented on frontend/i18n/locales/en.json:
```diff
@@ -6,7 +6,9 @@
         "beaconchain_homepage": "Beaconchain Homepage",
         "common": {
             "close": "Close",
+            "error_retry": "An error occurred. Please try again.",
```
**issue**: in the template you are referencing : `error_try_again`

enzo-bitfly replied:
thanks

marcel-bitfly commented on frontend/layers/base/app/components/BaseSearchInput.vue:
```diff
@@ -0,0 +1,168 @@
+<template>
+  <form
+    role="search"
+    class="p-2xl dark:bg-[rgba(16,16,16,0.2)] rounded-3xl"
```
**question:** why is `dark:bg-[rgba(16,16,16,0.2)]` not part of the `design system`?

Maybe we could give feedback to the designers.

enzo-bitfly replied:
It's the `--color-black` with 0.2 opacity. I changed it now to use a `:before`, which caused me to refactor the `index` page with div used for background effects.

marcel-bitfly commented on frontend/layers/base/app/components/BaseSearchInput.vue:
```diff
@@ -0,0 +1,168 @@
+        {{ label }}
+      </RkLabel>
+      <RkComboboxInput
+        id="blockchain-search-input"
```
**question**: wouldn't `search-input` be specific enough for a `BaseSearchInput` component?
```suggestion
        id="search-input"
```

enzo-bitfly replied:
good catch, missed that when refactoring

marcel-bitfly commented on frontend/layers/base/app/components/BaseSearchInput.vue:
```diff
@@ -0,0 +1,168 @@
+        class="search-input w-full text-2xl font-semibold rounded-3xl pt-3xl pr-5xl pb-6xl pl-2xl dark:bg-gray-950
+        dark:focus:bg-black placeholder:dark:text-gray-500 dark:text-white border-1 dark:border-charcoal-500
+        dark:focus:border-charcoal-50 dark:focus-within:outline-0"
+        @update:model-value="(value) => { if (!value) hasSearched = false }"
```
**np**: probably better to use a function here, since it might be interessting where `hasSearched` would be `reset`

marcel-bitfly commented on frontend/layers/base/app/components/BaseSearchInput.vue:
**thought:**since this seems not to be a generic component, probably it is best to place it in `products layer`

enzo-bitfly replied:
It is generic

marcel-bitfly commented on frontend/layers/base/app/composables/useBaseUrls.ts:
```diff
@@ -0,0 +1,15 @@
+export const useBaseUrls = () => {
```
**thought**: `baseUrl` normally refers to the `url` your `app` is deployed to

maybe something like `useEndpoint` would be clearer?

enzo-bitfly replied:
True. Calling it `useApiUrl` now, as endpoint is the specific endpoints in the `API`

marcel-bitfly commented on frontend/layers/products/app/components/BlockchainSearchInput.vue:
```diff
@@ -0,0 +1,79 @@
+</script>
+
+<template>
+  <BaseSearchInput
```
**thought**: maybe this is a bad name, is `BaseSearchInput` to generic enough to be used for any `Search` in our application no matter what `results` are expected?

enzo-bitfly replied:
That's the idea, yes: it's a combobox which displays search results passed in it's `results` prop. We may need to adapt where styles are applied later if another `search` has e.g. a different input style, but the functionality is agnostic.
One thing I missed is ensuring that the results are an array of objects, as that is a requirement of the component to work correctly. Added now: `generic="SearchResultType extends Record<string, any>"`

enzo-bitfly commented on frontend/layers/products/app/components/BlockchainSearchInput.vue:
```diff
@@ -0,0 +1,79 @@
+export type SearchResult = ReplaceChainId<InternalPostSearchResponse['data'][number]>
+type ReplaceChainId<T> = T extends { chain_id: any }
+  ? Omit<T, 'chain_id'> & { chain_id: ChainId }
+  : T
```
Forgot to replace this with the `InternalPostSearchResponseWithChainId` type from the `search` server route

marcel-bitfly commented on frontend/layers/products/app/components/BlockchainSearchResultItem.vue:
```diff
@@ -0,0 +1,125 @@
+    {{ result.value.address.hash }}
+  </NuxtLink>
+
+  <NuxtLink
```
**suggestion:** you could think about either `prefetch-on="interaction"` or `no-prefetch` for all your `NuxtLinks` here

as probably this would otherwise result in a lot of unnecessary traffic for the users

enzo-bitfly replied:
There's no prefetching happening, as the links lead to pages in a different application.

enzo-bitfly reviewed (commented):
Great review, thank you

marcel-bitfly commented on frontend/layers/base/app/utils/networks.ts:
```diff
@@ -0,0 +1,11 @@
+const ChainIDs = {
```
**np:** maybe we comply with the js convention for `constants`? 
```suggestion
const CHAIN_ID = {
```

enzo-bitfly replied:
Good call. Done

enzo-bitfly committed:
chore: install `reka-ui`

upgrade `nuxt` to prevent build error caused by `reka-ui` - see https://github.com/nuxt/ui/issues/4900

See: FE-92

enzo-bitfly committed:
feat: add `search` section to `landing page`

See: FE-92

enzo-bitfly committed:
fix(TheFooter): `api` link url

See: FE-99
